File: solutions/ocaml/difference-of-squares/1/difference_of_squares.ml
let square_of_sum n = 
    (of_int (n * (n + 1)/2)) ** 2.;;

let sum_of_squares n =
    n * (n + 1) * (2 * n + 1) / 6;;

let difference_of_squares n =
    square_of_sum n - sum_of_squares n ;;

File: solutions/c/difference-of-squares/1/difference_of_squares.c
#include "difference_of_squares.h"

int square_of_sum(int number)
{
    int sum = 0;
    for (int i = 1; i <= number; i++)
    {
        sum += i;
    }
    return sum * sum;
}

int sum_of_squares(int number)
{
    int sum = 0;
    for (int i = 1; i <+ number; i++)
    {
        sum += i * i;
    }
}

int difference_of_squares(int number)
{
    return square_of_sum(number) - sum_of_squares(number);
}